pkg/domain/infra/abi/images.go

```diff
@@ -759,11 +759,14 @@ func putSignature(manifestBlob []byte, mech signature.SigningMechanism, sigStore
 			return err
 		}
 	}
-	sigFilename, err := getSigFilename(signatureDir)
-	if err != nil {
-		return err
+	sigFileName := options.SigFile
+	if len(sigFileName) == 0 {
+		sigFileName, err = getSigFilename(signatureDir)
```

Comment:
If `signatureDir` is not used, this probably shouldn’t go through the trouble of computing it, and more importantly shouldn’t have the side-effect of creating a per-image directory.